packages/core-react/index.d.ts

```diff
@@ -9,3 +9,29 @@ export type TextFieldProps = {
 declare const TextField: React.FC<TextFieldProps>;
 
 export { TextField };
+
+export type ContainerProps = {
+  Tag?: React.ElementType<any>;
```

Comment:
following type from here: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/blob/master/ADVANCED.md#as-props-passing-a-component-to-be-rendered

Note: I am following the convention in this repo of naming the prop `Tag`. I do like the `as` prop naming convention for passed in tags. I can update if that is preferred.

Reply:
@adamraider any strong opinions about `as` vs. `Tag` prop?